docs/src/advanced/1_variants.md

```diff
@@ -0,0 +1,267 @@
+
+# Screening model variants
+
+A major goal of COBREXA is to make exploring of many model variants easy and
+fast.
+
+One main concept that can be utilized for doing that is implemented in function
+[`screen`](@ref), which takes your model, a list of model _variants_ that you
+want to explore by some specified _analysis_, and schedules the analysis of the
+model variants on all available distributed workers.
+
+In the most basic form, using the slightly simplified variant of
+[`screen`](@ref) that is called [`screen_variants`](@ref) may work as follows:
+
+```julia
+m = load_model(StandardModel, "ecoli_core_model.json")
+
+screen_variants(
+    m,    # the model for screening
+    [
+        [],    # a variant with no modifications
+        [with_set_bound("CO2t", lb = 0, ub = 0)],   # disable CO2 transport
+        [with_set_bound("O2t", lb = 0, ub = 0)],   # disable O2 transport
+    ],
+    m -> flux_balance_analysis_dict(m, Tulip.Optimizer)["BIOMASS_Ecoli_core_w_GAM"],
+)
+```
+The call specifies a model (the `m` that we have loaded) that is being tested,
+then a vector of model variants to be created and tested, and then the analysis
+that is being run on each variant -- in this case, we find an optimal steady
+state of each of the variants, and check out the biomass production rate at
+that state. In this particular case, we are checking what will be the effect of
+disabling CO2 transport and O2 transport in the cells. For that, we get the
+following result:
+```
+3-element Vector{Float64}:
+ 0.8739215022674809
+ 0.46166961413944896
+ 0.21166294973372796
+```
+
+The numbers are the biomass production rates for all variants. We can see that
+disabling O2 transport really does not help the organism much.
+
+## Variant specification
+
+In the above example, we have specified 3 variants, thus the analysis returned
+3 different results that correspond with the specifications. Let us have a look
+at the precise format of the specification and result.
+
+Importantly, the `variants` argument is of type `Array{Vector{Any}}`, meaning
+that it can be array of any dimensionality that contains vectors. Each of the
+vectors specifies precisely one variants, possibly with more modifications
+applied to the model in sequence.
+
+For example:
+- `[]` specifies no modifications at all
+- `[with_set_bound("CO2t", lb=0, ub=10)]` limits the CO2 transport
+- `[with_set_bound("CO2t", lb=0, ub=2), with_set_bound("O2t", lb=0, ub=100)]`
```

Comment:
ah, you do the multiple thing here, probably no need for PFL then

Reply:
Nah, good to repeat that.